rm_reporting/controllers/case_controller.py

```diff
@@ -25,8 +27,13 @@ def get_case_data(collection_exercise_id: str) -> list:
         "ORDER BY sample_unit_ref, status"
     )
 
-    with case_engine.begin() as conn:
-        return conn.execute(case_business_ids_query, {"collection_exercise_id": collection_exercise_id}).all()
+    try:
+        with case_engine.begin() as conn:
+            result = conn.execute(case_business_ids_query, {"collection_exercise_id": collection_exercise_id}).all()
+    except SQLAlchemyError:
+        logger.error("Failed to get case data", collection_exercise_id=collection_exercise_id)
+        abort(404, "Case data not found")
```

Comment:
I ended up replacing these `abort()` statements with a generic `return` statement because the `abort()` statement were making it difficult/impossible to create unit tests for. The behaviour of the code is otherwise the same; it still errors in the same way. Plus, the actual response code used for the abort statement didn't really matter because r-ops displays errors using a generic 500 page.

Reply:
It shouldn't matter if rOps displays a generic 500 page we should still return an appropriate status code and message so that if we decide to improve the response handling in rOps to give better information to our users (which we should do) we can use the information we have returned